geoshapes.js

```diff
@@ -240,7 +240,7 @@ GeoShapes.prototype.runWikidataQuery = function runWikidataQuery (xClientIp) {
         },
         headers: Object.assign(config.sparqlHeaders, { 'X-Client-IP': xClientIp })
     }).then(queryResult => {
-        if (queryResult.headers['content-type'] !== 'application/sparql-results+json') {
+        if (queryResult.headers['content-type'].indexOf('application/sparql-results+json') === -1) {
```

Comment:
I'd rather do:
```
if (!queryResult.headers['content-type'].startsWith('application/sparql-results+json'))
```
Even though practically the difference is not that important, this looks a bit cleaner.

Reply:
+1, and also the proposed change as written allows 'application/sparql-results+json' to be present anywhere in the header value, not just at the beginning, which isn't what we want.

Reply:
My thought was about performance, see http://jsben.ch/RVYk7 , to have the expected behavior in the beginning we could do:
```
if (queryResult.headers['content-type'].indexOf('application/sparql-results+json') !== 0)
```

But I am fine with `startsWith` too, it's more readable.

Reply:
I used to err on the side of the fastest-executing possible code, too, but I think the consensus in the software engineering community is that readability trumps sheer performance in general, because the impact of choices like these is negligible compared to things like I/O or a poorly thought out algorithm that ends up being slow.  (In this case, for example, the performance impact of our choice of startsWith vs. indexOf is dwarfed by the performance impact of the network call to WDQS we're about to make; so we might as well make the more readable choice.)

"Premature optimization" (discussed usefully [here](https://en.wikipedia.org/wiki/Program_optimization#When_to_optimize)) is the term of abuse that a lot of people apply to making the more performant but less readable choice.  I don't particularly care for that phrase, or think it's really _that_ bad, but I've come to prefer readability in general for the reason I mentioned above.

Reply:
Nice reading, thanks!